drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java

```diff
@@ -101,13 +94,15 @@ protected ZonedDateTime extractModificationDate(HttpClient.Response response) {
     public ExternalBinary getReleaseForVersion(String version) throws Exception {
         String url = projectUrl + RELEASES_URL;
         int pageNumber = 1;
+        List<String> availableVersions = new ArrayList<>();
 
         while (true) {
             HttpClient.Response response = sentGetRequestWithPagination(url, pageNumber++, Collections.emptyMap());
             JsonElement releases = gson.fromJson(response.getPayload(), JsonElement.class);
 
             if (releases != null && releases.isJsonArray() && releases.getAsJsonArray().size() > 0) {
-                ExternalBinary releaseForVersion = getReleaseForVersion(version, releases.getAsJsonArray());
+                ExternalBinary releaseForVersion =
```

Comment:
We can also extract the condition in `while` to a method with some helpful name.